include/deal.II/base/vectorization.h

```diff
@@ -656,6 +656,16 @@ class VectorizedArray
     base_ptr[offsets[0]] = data;
   }
 
+  /**
+   * Returns sum over entries of data field.
+   */
+  DEAL_II_ALWAYS_INLINE
+  Number
+  sum()
+  {
+    return data;
+  }
+
```

Comment:
What about moving down the implementations to the other ones?

Reply:
Do you think it is more intuitive to put it to the other ones? I kept it here as this is the special case with `width == 1` where nothing has to be done except returning `data`.

Reply:
These classes are a bit of outliers compared to the rest of deal.II because they contain many very short implementations and that all implementations are done at the place of declaration, rather than out-of-line definitions further down as in most other places in deal.II. I suggest to move the implementations in-line if possible in terms of the dependencies AVX-512 -> AVX -> SSE2, otherwise it would be good to move the implementation down. We have a bigger topic to tackle in terms of #11719.

Reply:
Inline implementation works if i arrange the specializations in a different order, see ba78f77f58ac656521e50a24ac3205b1314d28b3 .